### ISSUE TITLE
Symbol list not fully support trailing return type in C++11.

### ISSUE THREAD

**Issue by ToyAuthor** [geany/geany#2509]
O:show on symbol list
X:just like never exist

auto getName(int id)->::std::string; // O
auto getName(int id)->std::string; // X
auto getName(int)->::std::string; // O
auto getName(int)->std::string; // X
auto getName(uint32_t id)->std::string; // X
auto getName(uint32_t)->std::string; // O

Use global namespace could cover this bug for now.

**Comment by elextr** [comment]
Yeah, the C++ parser is pretty old, upstream uctags has a completly new one, but "somebody" has to integrate it into Geany.
